test/http_server_test/http_server_test.cpp

```diff
@@ -220,112 +230,131 @@ namespace http_server_test
-                                                                                                                                                                                            +
-                    uploads.str()), false);
-                                               }
-                                           };
-
-                          auto url_encoded_data =
-                              [&response, &last_part](std::unordered_map<std::string, std::string>& data) {
-                                  if (last_part)
-                                  {
-                                      response.reply(std::make_unique<responses::StringResponse>(
+                IServerResponse& response,
+                IConnectionTimeoutModifier& timeout_modifier,
+                const std::string& url,
+                bool first_part,
+                bool last_part,
+                const std::unordered_map<std::string, std::string>& headers,
+                const std::unordered_map<std::string, std::string>& request_parameters,
+                const std::vector<uint8_t>& content,
+                MIMEParser& mime) 
+        {
+            (void)timeout_modifier;
+            (void)request_parameters;
+            (void)url;
+
+            if (first_part)
+            {
+                // Prepare mime parser to receive data
+                mime.detect_mode(headers.at(CONTENT_TYPE), std::stoul(headers.at(CONTENT_LENGTH)));
+            }
+
+            auto form_data = [uploads, &last_part, &response](const std::string& name,
+                    const std::string& actual_file_name,
+                    const MIMEParser::BoundaryIterator& begin,
+                    const MIMEParser::BoundaryIterator& end,
+                    const bool file_start,
+                    const bool file_close)  
+            {
+                (void) name;
+                Path path{};
+                static std::ofstream to_save;
+                if(file_start)
+                {
+                    // Store the file in web_root/uploads
+                    path = uploads / actual_file_name;
+                    Log::info("form_data", "File name: {}", path.str());
+                    create_directory(path.parent());
+                    if (FileInfo{ path }.exists())
+                    {
+                        Log::info("form_data", "File exists");
+                        remove(path);
+                    }
+                    Log::info("form_data", "before open");
+                    to_save.open(path.str(), std::ios::out | std::ios::app | std::ios::binary) ;
+                    Log::info("form_data", "File opened");
+                }
+
+                auto len = std::distance(begin, end);
+                Log::info("form_data", "chunk to write: {},", static_cast<int>(len));
+                to_save.write((char*)&*begin, static_cast<int>(len));
```

Comment:
Had forgotten about you asking about this one. Try this, it compiles on both for Linux and Xtensa.

`to_save.write(reinterpret_cast<const char*>(&*begin), static_cast<int>(len));`